vm/interface.go

```diff
@@ -31,15 +31,30 @@ type SystemSCContainer interface {
 
 // SystemEI defines the environment interface system smart contract can use
 type SystemEI interface {
+	ExecuteOnDestContext(destination []byte, sender []byte, value *big.Int, input []byte) (*vmcommon.VMOutput, error)
 	Transfer(destination []byte, sender []byte, value *big.Int, input []byte) error
 	GetBalance(addr []byte) *big.Int
 	SetStorage(key []byte, value []byte)
 	GetStorage(key []byte) []byte
-	SelfDestruct(beneficiary []byte)
 	Finish(value []byte)
 	BlockChainHook() vmcommon.BlockchainHook
 	CryptoHook() vmcommon.CryptoHook
 
+	IsInterfaceNil() bool
+}
+
+// ContextHandler defines the methods needed to execute system smart contracts
+type ContextHandler interface {
```

Comment:
Why do we need this interface to be an extension of SystemEI interface? Why not make a real extension instead of redefining each method.

Reply:
done.